fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -576,6 +576,7 @@ public CommandProcessingResult disburseLoan(final Long loanId, final JsonCommand
                     // the existing loan
                     amountToDisburse = disburseAmount.minus(loanOutstanding);
                 }
+                createRestructuringCancellationEvent(loanToClose); // Generate the event
```

Comment:
@oluexpert99 If you look at line 574, it shows that top up functionality is also being used when a write-off Punish (castigado) action is performed. Your changes will also create the Restructuring news for castigado action which is not mentioned in the ticket

Reply:
Hi @faheem205  , please check again 
there is an entry to be sure the loan is a top up loan 

if (loan.isTopup() && loan.getClientId() != null) {

Reply:
@oluexpert99 Yes. When doing a castigado topup field is set to true so that the old loan can be closed and new loan created. If the loan is not marked as a topup loan then castigado check does not makes sense inside the if statement. Please update your code